components/unity/unity_port_esp32.c

```diff
@@ -28,6 +29,37 @@ void unity_flush(void)
     esp_rom_uart_tx_wait_idle(CONFIG_ESP_CONSOLE_UART_NUM);
 }
 
+#define iscontrol(c) ((c) <= '\x1f' || (c) == '\x7f')
+
+static void esp_unity_readline(char* dst, size_t len)
+{
+    /* Read line from console with support for echoing and backspaces */
+    size_t write_index = 0;
+    for (;;) {
+        char c = 0;
+        bool got_char = esp_rom_uart_rx_one_char((uint8_t*)&c) == 0;
+        if (!got_char) {
+          continue;
+        }
+        if (c == '\r' || c == '\n') {
+            unity_putc('\n');
+            dst[write_index] = '\0';
+            return;
+        } else if (c == '\b') {
+            if (write_index > 0) {
+                write_index--;
+                // Delete previously entered character
+                esp_rom_uart_tx_one_char('\b');
+                esp_rom_uart_tx_one_char(' ');
+                esp_rom_uart_tx_one_char('\b');
+            }
+        } else if (write_index < len - 1 && !iscontrol(c)) {
+            unity_putc(c);
+            dst[write_index++] = c;
+        }
```

Comment:
It would likely be a good idea to add:
```
} else if (write_index >= len) {
  return;
}
```
That way it can exit from this loop without over running the buffer.

Reply:
I don't believe this code will overrun the buffer (though I may be mistaken). 
It only increments `write_index` and writes to `dst` if it's less than `len - 1` (to allow space for a null terminator in the return). I don't return when the buffer is full as it's still possible to backspace to delete the contents.

Reply:
It would overflow in one case, len = 0 (since len - 1 would underflow, which is now fixed)